src/memory_map.rs

```diff
@@ -13,7 +13,9 @@ impl MemoryMapTag {
         let start_area = (&self.first_area) as *const MemoryArea;
         MemoryAreaIter {
             current_area: start_area,
-            last_area: ((self_ptr as u32) + self.size - self.entry_size) as *const MemoryArea,
+            last_area: unsafe {
+                self_ptr.offset((self.size - self.entry_size) as isize) as *const MemoryArea
```

Comment:
The argument of the [offset](https://doc.rust-lang.org/nightly/std/primitive.pointer.html#method.offset) method is not in units of bytes:

> `unsafe fn offset(self, count: isize) -> *const T`
> Calculates the offset from a pointer. `count` is in units of `T`; e.g. a `count` of 3 represents a pointer offset of `3 * sizeof::<T>()` bytes.

So this change would lead to a wrong pointer.